cime/config/e3sm/machines/config_machines.xml

```diff
@@ -2949,17 +2674,14 @@
         <command name="load">pgi/19.9</command>
       </modules>
       <modules compiler="pgigpu">
-        <command name="load">pgi/19.4</command>
+        <command name="load">pgi/19.9</command>
         <command name="load">cuda</command>
```

Comment:
PGI 19.10 is tested nightly on Compy. Can this load `pgi/19.10`?
It'd be good to add a specific version like `cuda/10.1.243` for reproducibility (default module versions change).

Reply:
I had a lot of difficulty getting the MMF GPU cases to run, but changing the PGI version to 19.9 seemed to matter a lot. I don't remember the details, but they probably don't matter because I remember getting a lot of garbage log files that didn't really point me to any specific issue. I can try again since we made some other changes.

Reply:
If Summit has to stay at 19.9, we should ask to have 19.9 installed on compy.  The point of PGI testing on compy is track/bakcup Summit.

Reply:
I prefer to updating PGI to 20.1 as all PGI 2019 versions has a problem to inline functions using the IPA option and LLVM as its backend.  But I am open to update PGI in this PR or create a new PR. I agree with @amametjanov to add cuda version number for reproducibility.

Reply:
ok, let me revisit the MMF-GPU case with the pgi/20.1 and cuda/10.1.243

Reply:
ok, so I ran some MMF SMS tests with pgi/19.9, both CPU and GPU, and those passed. Then I switched to pgi/20.1 and I get this build error:
```
3698222:PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (4080,base=4083) member symbol shr_infnan_nan$ac (offset=61): not found!  (/autofs/nccs-svm1_home1/hannah6/E3SM/E3SM_SRC2/components/clm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
3698491:PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (4080,base=4083) member symbol shr_infnan_snan$ac (offset=63): not found!  (/autofs/nccs-svm1_home1/hannah6/E3SM/E3SM_SRC2/components/clm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
3698761:PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (4080,base=4083) member symbol shr_infnan_qnan$ac (offset=65): not found!  (/autofs/nccs-svm1_home1/hannah6/E3SM/E3SM_SRC2/components/clm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
3699031:PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (4080,base=4083) member symbol shr_infnan_inf$ac (offset=67): not found!  (/autofs/nccs-svm1_home1/hannah6/E3SM/E3SM_SRC2/components/clm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
3699300:PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (4080,base=4083) member symbol shr_infnan_posinf$ac (offset=69): not found!  (/autofs/nccs-svm1_home1/hannah6/E3SM/E3SM_SRC2/components/clm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
3699572:PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (4080,base=4083) member symbol shr_infnan_neginf$ac (offset=71): not found!  (/autofs/nccs-svm1_home1/hannah6/E3SM/E3SM_SRC2/components/clm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
3699844:PGF90-F-0000-Internal compiler error. interf:new_symbol_and_link, symbol not found    2568  (/autofs/nccs-svm1_home1/hannah6/E3SM/E3SM_SRC2/components/clm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
3700062-PGF90/power Linux 20.1-0: compilation aborted
3700108-gmake[2]: *** [cmake/lnd/CMakeFiles/lnd.dir/__/__/clm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90.o] Error 2
```
The submodules are up to date, so it appears we'll need to get the land people to fix this if we want to use 20.1.

Reply:
@whannah1 Thanks a lot for testing PGI/20.1 It is an ICE error, we may need to report to the OLCF, so PGI engineers could work on it.

Reply:
Don't wait to get PGI 20.1 working for this PR to go forward.  You can update it later.

Reply:
ok, sounds good, then this PR should be ready to go.

Reply:
No.  Remove the GNU changes.  Add an explicit version to the cuda module load.

Reply:
oh right, I'll run another test with cuda/10.1.243

Reply:
Test passed!